### PR TITLE
Login/popup window and FreeResource styling

### PR DESCRIPTION
**Pull request by Milutin25**
Created login modal for porch-check your stats button(when clicked login modal shows).
Implemented this wherever user is asked to be logged in. 

Created component in Layout/CardsLayout for styling cards on Porch Page and FreeResources Page, that is reusable. Still need to work on button on FreeRes page and to do some additional styling. 

How to Test? 
- .....

### CODE REVIEW

File: components/FreeSourcesList/index.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { FreeSource } from "./FreeSource";
 
 interface FreeSourcesListProps {
   facts: any[];
-  setFacts: any;
+  setFacts: any;
```
Comment (slavo3dev) [inline components/FreeSourcesList/index.tsx:7]: we need to fix this , we can not do any

File: components/Layout/CardsLayout/index.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    excellent?: number;
+  };
+  displayComment: string;
+  commentText: string;
+  showMore?: boolean;
+  handleMore?: () => void;
+  handleVote: (type: string) => void;
+  isUpdating: boolean;
+  formattedDate?: string;
+  extraContent?: React.ReactNode;
```
Comment (slavo3dev) [inline components/Layout/CardsLayout/index.tsx:19]: import ReactNode

File: components/PorchElements/PorchHeader/index.tsx [hunk 3]
```diff
@@ -1,10 +1,10 @@
-import React, { useContext } from "react";
+import React, { useContext, useState, useEffect } from "react";
```
Comment (slavo3dev) [inline components/PorchElements/PorchHeader/index.tsx:1]: remove react

**Review by slavo3dev — CHANGES_REQUESTED** [review]
Small changes , lGTM

**Review by SdZika — APPROVED** [review]
All good.

**Review by slavo3dev — APPROVED** [review]

**Pull request merged**